src/renderer.cpp

```diff
@@ -1357,6 +1750,248 @@ void Renderer::setD3DProjection(struct matrix* matrix)
+}
+
+float Renderer::getRoughness()
+{
+    return internalState.materialData[0];
+}
+
+void Renderer::setMetalness(float metalness)
+{
+    internalState.materialData[1] = metalness;
+}
+
+float Renderer::getMetalness()
+{
+    return internalState.materialData[1];
+}
+
+void Renderer::setShadowFaceCullingEnabled(bool isEnabled)
+{
+    isShadowMapFaceCullingEnabled = isEnabled;
+}
+
+bool Renderer::isShadowFaceCullingEnabled()
+{
+    return isShadowMapFaceCullingEnabled;
+}
+
+void Renderer::setShadowMapResolution(int resolution)
+{
+    internalState.shadowData[3] = resolution;
+
+    if (bgfx::isValid(shadowMapFrameBuffer))
+    {
+        bgfx::destroy(shadowMapFrameBuffer);
+    }
+
+    shadowMapTexture = bgfx::createTexture2D(
+        resolution,
+        resolution,
+        false,
+        1,
+        bgfx::TextureFormat::D32,
+        BGFX_TEXTURE_RT | BGFX_SAMPLER_COMPARE_LEQUAL | BGFX_SAMPLER_U_CLAMP | BGFX_SAMPLER_V_CLAMP | BGFX_SAMPLER_W_CLAMP
+    );
+
+    shadowMapFrameBuffer = bgfx::createFrameBuffer(
+        1,
+        &shadowMapTexture,
+        true
+    );
+}
+
+int Renderer::getShadowMapResolution()
+{
+    return internalState.shadowData[3];
+}
+
+void Renderer::setShadowConstantBias(float bias)
+{
+    internalState.shadowData[0] = bias;
+}
+
+float Renderer::getShadowConstantBias()
+{
+    return internalState.shadowData[0];
+}
+
+void Renderer::setShadowMapArea(float area)
+{
+    shadowMapArea = area;
+}
+
+float Renderer::getShadowMapArea()
+{
+    return shadowMapArea;
+}
+
+void Renderer::setShadowMapNearFarSize(float size)
+{
+    shadowMapNearFarSize = size;
+}
+
+float Renderer::getShadowMapNearFarSize()
+{
+    return shadowMapNearFarSize;
+}
+
+void Renderer::setFieldShadowMapArea(float area)
+{
+    fieldShadowMapArea = area;
+}
+
+float Renderer::getFieldShadowMapArea()
+{
+    return fieldShadowMapArea;
+}
+
+void Renderer::setFieldShadowMapNearFarSize(float size)
+{
+    fieldShadowMapNearFarSize = size;
+}
+
+float Renderer::getFieldShadowMapNearFarSize()
+{
+    return fieldShadowMapNearFarSize;
+}
+
+void Renderer::setFieldShadowOcclusion(float value)
+{
+    internalState.fieldShadowData[0] = value;
+}
+
+float Renderer::getFieldShadowOcclusion()
+{
+    return internalState.fieldShadowData[0];
+}
+
+void Renderer::setFieldShadowFadeStartDistance(float value)
+{
+    internalState.fieldShadowData[1] = value;
+}
+
+float Renderer::getFieldShadowFadeStartDistance()
+{
+    return internalState.fieldShadowData[1];
+}
+
+void Renderer::setFieldShadowFadeRange(float value)
+{
+    internalState.fieldShadowData[2] = value;
+}
+
+float Renderer::getFieldShadowFadeRange()
+{
+    return internalState.fieldShadowData[2];
+}
+
+void Renderer::setWalkmeshExtrudeSize(float size)
+{
+    walkMeshExtrudeSize = size;
+}
+
+float Renderer::getWalkmeshExtrudeSize()
+{
+    return walkMeshExtrudeSize;
+}
+
+void Renderer::setWalkmeshPosOffset(float offset)
+{
+    walkMeshPosOffset = offset;

```

Comment:
Shouldn't this be part of the `internalState`?

Reply:
I moved this variable to the `internalState`.